packages/test/test-end-to-end-tests/src/test/waitForSummary.spec.ts

```diff
@@ -0,0 +1,220 @@
+	type IHostLoader,
+} from "@fluidframework/container-definitions/internal";
+import type { IContainerExperimental } from "@fluidframework/container-loader/internal";
+import type {
+	ConfigTypes,
+	IConfigProviderBase,
+	IRequestHeader,
+} from "@fluidframework/core-interfaces";
+import type { ISharedMap } from "@fluidframework/map/internal";
+import {
+	type ITestObjectProvider,
+	type ITestContainerConfig,
+	createSummarizer,
+	summarizeNow,
+	type ChannelFactoryRegistry,
+	createAndAttachContainer,
+	DataObjectFactoryType,
+	type ITestFluidObject,
+	waitForContainerConnection,
+	timeoutAwait,
+} from "@fluidframework/test-utils/internal";
+
+// eslint-disable-next-line import/no-internal-modules
+import { loadOffline, getPendingOps } from "./offline/offlineTestsUtils.js";
+
+const waitForSummary = async (
+	provider: ITestObjectProvider,
+	container: IContainer,
+	testContainerConfig: ITestContainerConfig,
+	summaryVersion?: string,
+) => {
+	const { summarizer, container: summarizingContainer } = await createSummarizer(
+		provider,
+		container,
+		testContainerConfig,
+		summaryVersion,
+	);
+	await provider.ensureSynchronized();
+	const result = await summarizeNow(summarizer);
+	summarizingContainer.close();
+	return result.summaryVersion;
+};
+
+const configProvider = (settings: Record<string, ConfigTypes>): IConfigProviderBase => ({
+	getRawConfig: (name: string): ConfigTypes => settings[name],
+});
+
+describeCompat(
+	"Offline tests that wait for a summary",
+	"NoCompat",
+	(getTestObjectProvider, apis) => {
+		const mapId = "map";
+		const { SharedMap } = apis.dds;
+		const registry: ChannelFactoryRegistry = [[mapId, SharedMap.getFactory()]];
+		let provider: ITestObjectProvider;
+		let loader: IHostLoader;
+		let container: IContainerExperimental;
+		let url: any;
+		let map1: ISharedMap;
+		let dataStore1: ITestFluidObject;
+		const testContainerConfig: ITestContainerConfig = {
+			fluidDataObjectType: DataObjectFactoryType.Test,
+			registry,
+			loaderProps: {
+				configProvider: configProvider({
+					"Fluid.Container.enableOfflineLoad": true,
+					"Fluid.Sequence.intervalStickinessEnabled": true,
+				}),
+			},
+		};
+		const mainContainerConfig: ITestContainerConfig = {
+			...testContainerConfig,
+			runtimeOptions: {
+				summaryOptions: {
+					summaryConfigOverrides: {
+						state: "disabled",
+					},
+				},
+			},
+		};
+
+		beforeEach("setup", async () => {
+			provider = getTestObjectProvider({ syncSummarizer: true });
+			loader = provider.makeTestLoader(mainContainerConfig);
+			container = await createAndAttachContainer(
+				provider.defaultCodeDetails,
+				loader,
+				provider.driver.createCreateNewRequest(provider.documentId),
+			);
+			provider.updateDocumentId(container.resolvedUrl);
+			url = await container.getAbsoluteUrl("");
+			dataStore1 = (await container.getEntryPoint()) as ITestFluidObject;
+			map1 = await dataStore1.getSharedObject<ISharedMap>(mapId);
+			map1.set("1", "1");
```

Comment:
Are you doing this to force a write connection?  Nice to put a comment to that effect (some places even do `set("force", "write connection");` 🤷‍♂️